src/System.Threading.Thread/tests/ThreadTests.netstandard.cs

```diff
@@ -20,10 +20,21 @@ public static class ThreadTests
         [Fact]
         public static void ConstructorTest()
         {
-            var t = new Thread(() => { });
-            t = new Thread(() => { }, 0);
-            t = new Thread(state => { });
-            t = new Thread(state => { }, 0);
+            Action<Thread> startThreadAndJoin =
+                t =>
+                {
+                    t.IsBackground = true;
+                    t.Start();
+                    t.Join();
+                };
+            startThreadAndJoin(new Thread(() => { }));
+            startThreadAndJoin(new Thread(() => { }, 0));
+            startThreadAndJoin(new Thread(() => { }, 1));
+            startThreadAndJoin(new Thread(() => { }, 16 << 20));
```

Comment:
I tried it but it looks far less readable. As I'd like the constructor call to be in the test and not in the member data generation, and I'm testing different constructors here, it doesn't seem worth having 4 separate tests of which 2 would be theories for something trivial like this. If this fails I'm fine with it stopping at the first failure.